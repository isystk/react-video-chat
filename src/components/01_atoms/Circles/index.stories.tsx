import { Meta, Story } from '@storybook/react'
import React from 'react'
import Circles from './index'

export default {
  title: '01_atoms/Circles',
  component: Circles,
} as Meta

const Template: Story = (props) => {
  return (
    <Circles>
      <div
        style={{
          width: '100%',
          height: '600px',
          background: '#eee',
        }}
      ></div>
    </Circles>
  )
}

export const Primary = Template.bind({})
Primary.storyName = 'プライマリ'
Primary.args = {}
