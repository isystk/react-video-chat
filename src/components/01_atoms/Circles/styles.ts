import { makeStyles } from '@material-ui/core/styles'

const useStyles = makeStyles(() => ({
  circles: {
    position: 'relative',
    top: '0',
    left: '0',
    width: '100%',
    height: '100%',
    overflow: 'hidden',
  },
  circlesLi: {
    position: 'absolute',
    display: 'block',
    listStyle: 'none',
    width: '20px',
    height: '20px',
    background: '#fff',
    animation: '$fadeOutRotation 25s linear infinite',
    bottom: '-150px',
  },
  '@keyframes fadeOutRotation': {
    '0%': {
      transform: 'translateY(0) rotate(0deg)',
      opacity: '1',
      borderRadius: '0',
    },
    '100%': {
      transform: 'translateY(-1000px) rotate(720deg)',
      opacity: '0',
      borderRadius: '50%',
    },
  },
}))

export { useStyles }
