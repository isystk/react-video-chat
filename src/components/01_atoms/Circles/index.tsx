import React, { FC } from 'react'
import { ContainerProps, WithChildren } from 'types'
import { connect } from '@/components/hoc'
import { useStyles } from './styles'

/** CirclesProps Props */
export type CirclesProps = WithChildren
/** Presenter Props */
export type PresenterProps = CirclesProps & { classes; getLiPos }

/** Presenter Component */
const CirclesPresenter: FC<PresenterProps> = ({
  children,
  classes,
  getLiPos,
  ...props
}) => (
  <>
    <ul className={classes.circles}>
      <li className={classes.circlesLi} style={getLiPos(0)}></li>
      <li className={classes.circlesLi} style={getLiPos(1)}></li>
      <li className={classes.circlesLi} style={getLiPos(2)}></li>
      <li className={classes.circlesLi} style={getLiPos(3)}></li>
      <li className={classes.circlesLi} style={getLiPos(4)}></li>
      <li className={classes.circlesLi} style={getLiPos(5)}></li>
      <li className={classes.circlesLi} style={getLiPos(6)}></li>
      <li className={classes.circlesLi} style={getLiPos(7)}></li>
      <li className={classes.circlesLi} style={getLiPos(8)}></li>
      <li className={classes.circlesLi} style={getLiPos(9)}></li>
      {children}
    </ul>
  </>
)

/** Container Component */
const CirclesContainer: React.FC<
  ContainerProps<CirclesProps, PresenterProps>
> = ({ presenter, children, ...props }) => {
  const classes = useStyles()

  const getLiPos = (index) =>
    [
      { left: '25%', width: '80px', height: '80px', animationDelay: '0s' },
      {
        left: '10%',
        width: '20px',
        height: '20px',
        animationDelay: '2s',
        animationDuration: '12s',
      },
      { left: '70%', width: '20px', height: '20px', animationDelay: '4s' },
      {
        left: '40%',
        width: '60px',
        height: '60px',
        animationDelay: '0s',
        animationDuration: '18s',
      },
      { left: '65%', width: '20px', height: '20px', animationDelay: '0s' },
      { left: '75%', width: '110px', height: '110px', animationDelay: '3s' },
      { left: '35%', width: '150px', height: '150px', animationDelay: '7s' },
      {
        left: '50%',
        width: '25px',
        height: '25px',
        animationDelay: '15s',
        animationDuration: '45s',
      },
      {
        left: '20%',
        width: '15px',
        height: '15px',
        animationDelay: '2s',
        animationDuration: '35s',
      },
      {
        left: '85%',
        width: '150px',
        height: '150px',
        animationDelay: '0s',
        animationDuration: '11s',
      },
    ][index]

  return presenter({
    children,
    classes,
    getLiPos,
    ...props,
  })
}

export default connect<CirclesProps, PresenterProps>(
  'Circles',
  CirclesPresenter,
  CirclesContainer
)
