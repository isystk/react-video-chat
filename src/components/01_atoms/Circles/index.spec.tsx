import React from 'react'
import renderer from 'react-test-renderer'
import Circles from './index'
import '@testing-library/jest-dom/extend-expect'

describe('Circles', () => {
  it('Match Snapshot', () => {
    const component = renderer.create(<Circles />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
