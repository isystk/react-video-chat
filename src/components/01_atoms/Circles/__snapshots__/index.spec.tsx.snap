// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circles Match Snapshot 1`] = `
<ul
  className="makeStyles-circles-1"
>
  <li
    className="makeStyles-circlesLi-2"
    style={
      Object {
        "animationDelay": "0s",
        "height": "80px",
        "left": "25%",
        "width": "80px",
      }
    }
  />
  <li
    className="makeStyles-circlesLi-2"
    style={
      Object {
        "animationDelay": "2s",
        "animationDuration": "12s",
        "height": "20px",
        "left": "10%",
        "width": "20px",
      }
    }
  />
  <li
    className="makeStyles-circlesLi-2"
    style={
      Object {
        "animationDelay": "4s",
        "height": "20px",
        "left": "70%",
        "width": "20px",
      }
    }
  />
  <li
    className="makeStyles-circlesLi-2"
    style={
      Object {
        "animationDelay": "0s",
        "animationDuration": "18s",
        "height": "60px",
        "left": "40%",
        "width": "60px",
      }
    }
  />
  <li
    className="makeStyles-circlesLi-2"
    style={
      Object {
        "animationDelay": "0s",
        "height": "20px",
        "left": "65%",
        "width": "20px",
      }
    }
  />
  <li
    className="makeStyles-circlesLi-2"
    style={
      Object {
        "animationDelay": "3s",
        "height": "110px",
        "left": "75%",
        "width": "110px",
      }
    }
  />
  <li
    className="makeStyles-circlesLi-2"
    style={
      Object {
        "animationDelay": "7s",
        "height": "150px",
        "left": "35%",
        "width": "150px",
      }
    }
  />
  <li
    className="makeStyles-circlesLi-2"
    style={
      Object {
        "animationDelay": "15s",
        "animationDuration": "45s",
        "height": "25px",
        "left": "50%",
        "width": "25px",
      }
    }
  />
  <li
    className="makeStyles-circlesLi-2"
    style={
      Object {
        "animationDelay": "2s",
        "animationDuration": "35s",
        "height": "15px",
        "left": "20%",
        "width": "15px",
      }
    }
  />
  <li
    className="makeStyles-circlesLi-2"
    style={
      Object {
        "animationDelay": "0s",
        "animationDuration": "11s",
        "height": "150px",
        "left": "85%",
        "width": "150px",
      }
    }
  />
</ul>
`;
